pkg/state/finalization.go

```diff
@@ -0,0 +1,85 @@
+package state
+
+import (
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+	"github.com/pkg/errors"
+	"github.com/wavesplatform/gowaves/pkg/proto"
+)
+
+const finalizationKey = "finalization"
+
+var ErrNoFinalization = errors.New("no finalized blocks recorded")
+var ErrNoFinalizationHistory = errors.New("no finalization in history")
+
+type finalizationItem struct {
+	Block                proto.BlockHeader `cbor:"0,keyasint,omitempty"`
+	FinalizedBlockHeight proto.Height      `cbor:"1,keyasint,omitempty"`
+}
+
+type finalizationRecord struct {
+	Records []finalizationItem `cbor:"0,keyasint,omitempty"`
```

Comment:
It's going to be a huge amount of data. Why store list of items in history record which in turn a list of changes on every block?